src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs

```diff
@@ -95,7 +95,7 @@ public static void UpdateRunSettingsNode(this IRunSettingsProvider runSettingsPr
         /// <returns></returns>
         public static Match GetTestRunParameterNodeMatch(this IRunSettingsProvider runSettingsProvider, string node)
         {
-            var attrName = $"(?<{AttributeNameString}>\\w+)";
+            var attrName = $"(?<{AttributeNameString}>(\\w+\\S*\\w+)|(\\w+))";
```

Comment:
Won't this match way more than we want, that is any non-whitespace, even repeated one?

Reply:
Maybe we should use `[\w.:-]+` matching `A-Z a-z 0-9 _ . : -` which would allow reasonably separating the values without imposing any structure to it? This string has no semantic meaning to us, so forcing structure seems like overkill. Especially since _ is already part of the `\w` group so you should be able to do `__________hello_this_is_dog__` already.

Reply:
Sounds good, I'll go ahead and changes this to `[\w.:-]+` then. Many thanks for your input ! 👍

Reply:
Done.